### PR TITLE
cherry-pick fixes needed for 1.12.2

### PR DESCRIPTION
**Pull request by sanimej**

### CODE REVIEW

File: sandbox.go [hunk 1]
```diff
@@ -437,7 +437,7 @@ func (sb *sandbox) ResolveIP(ip string) string {
 }
 
 func (sb *sandbox) execFunc(f func()) {
```
Comment (mavenugo) [inline sandbox.go]: Please add error return type to this function.

File: resolver.go [hunk 2]
```diff
@@ -354,8 +354,8 @@ func (r *resolver) ServeDNS(w dns.ResponseWriter, query *dns.Msg) {
 				extConn, err = net.DialTimeout(proto, addr, extIOTimeout)
 			}
 
-			r.sb.execFunc(extConnect)
-			if err != nil {
+			execErr := r.sb.execFunc(extConnect)
```
Comment (mavenugo) [inline resolver.go]: compilation error due to missing error return

Comment (sanimej) [inline resolver.go]: @mavenugo fixed it. PTAL

**Comment by mavenugo** [comment]
thanks @sanimej . LGTM

**Pull request merged**